src/container/container.ts

```diff
@@ -502,69 +500,68 @@ class Container implements interfaces.Container {
     // Prepares arguments required for resolution and
     // delegates resolution to _middleware if available
     // otherwise it delegates resolution to _planAndResolve
-    private _get<T>(defaultArgs: interfaces.NextArgs): (T | T[] | Promise<T> | Promise<T>[]) {
+    private _get<T>(getArgs: GetArgs): (T | T[] | Promise<T> | Promise<T>[]) {
         let result: (T | T[] | Promise<T> | Promise<T>[]) | null = null;
-
+        const planAndResolveArgs:interfaces.NextArgs = {
+            ...getArgs,
+            contextInterceptor:(context) => context,
+            targetType: TargetTypeEnum.Variable
+        }
         if (this._middleware) {
-            result = this._middleware(defaultArgs);
+            result = this._middleware(planAndResolveArgs);
             if (result === undefined || result === null) {
                 throw new Error(ERROR_MSGS.INVALID_MIDDLEWARE_RETURN);
             }
         } else {
-            result = this._planAndResolve<T>()(defaultArgs);
+            result = this._planAndResolve<T>()(planAndResolveArgs);
         }
 
         return result;
     }
 
-    private _getWithAsyncContext<T>(
-        isAsync: boolean,
-        defaultArgs: interfaces.NextArgs,
+    private _getButThrowIfAsync<T>(
+        getArgs: GetArgs,
     ): (T | T[] | Promise<T> | Promise<T>[]) {
-        const result = this._get<T>(defaultArgs);
+        const result = this._get<T>(getArgs);
 
-        if (isPromise(result) && !isAsync) {
-            throw new Error(ERROR_MSGS.LAZY_IN_SYNC(defaultArgs.serviceIdentifier));
+        if (isPromise(result)) {
```

Comment:
if result is `(T | Promise<T>)[]` we could miss the throw, I know how to recreate this and fix it :)

Reply:
Spotted in - .some test again